src/script/interpreter.cpp

```diff
@@ -493,6 +493,26 @@ static bool CheckSignatureEncodingSigHashChoice(const vector<unsigned char> &vch
     {
         return true;
     }
+
+    if (flags & SCRIPT_ENABLE_SCHNORR)
+    {
+        if (vchSig.size() == 64 + (check_sighash == true) ? 1 : 0) // 64 sig length plus 1 sighashtype
+        {
+            // In a generic-signature context, 64-byte signatures are interpreted
+            // as Schnorr signatures (always correctly encoded) when flag set.
+            if (check_sighash && ((flags & SCRIPT_VERIFY_STRICTENC) != 0))
+            {
+                if (!IsDefinedHashtypeSignature(vchSig))
+                    return set_error(serror, SCRIPT_ERR_SIG_HASHTYPE);
+
+                // schnorr sigs must use forkid sighash if forkid flag set
+                if ((flags & SCRIPT_ENABLE_SIGHASH_FORKID) && ((vchSig[64] & SIGHASH_FORKID) == 0))
```

Comment:
and here maybe:
`if ((flags & SCRIPT_ENABLE_SIGHASH_FORKID) && (!(vchSig[64] & SIGHASH_FORKID)))`

Reply:
as above